### PR TITLE
block: Skip cold-plugging block devices.

### PR DESCRIPTION
**Pull request by amshinde**
Since vfio devices cannot be hotplugged currently, we
need to call atachDevices from pod.go.
We can avoid cold plugging block devices with this check for now.

Fixes https://github.com/clearcontainers/runtime/issues/802

Signed-off-by: Archana Shinde <archana.m.shinde@intel.com>

### CODE REVIEW

**Comment by sboeuf** [comment]
It's all good !!
LGTM

**Comment by sameo** [comment]
LGTM

File: device.go [hunk 1]
```diff
@@ -187,6 +187,15 @@ func makeBlockDevIDForHypervisor(deviceID string) string {
 }
 
 func (device *BlockDevice) attach(h hypervisor, c *Container) (err error) {
```
Comment (mcastelino) [inline device.go]: @devimc @amshinde are we doing this only for block devices. Can we do this for network devices also?

Comment (sboeuf) [reply device.go]: This PR only intends to fix https://github.com/clearcontainers/runtime/issues/802
We have discussed that @devimc is gonna submit a more global PR for removing all the coldplug codepaths (see my comment on #471). This PR will be blocked until we move to qemu2.9

Comment (mcastelino) [reply device.go]: fair enough.

**Comment by devimc** [comment]
lgtm

**Comment by jodh-intel** [comment]
lgtm

**Pull request merged**